shared/libraries/opcuatms/opcuatms_client/src/objects/tms_client_component_impl.cpp

```diff
@@ -12,20 +12,34 @@ using namespace daq::opcua;
 
 template <class Impl>
 ErrCode TmsClientComponentBaseImpl<Impl>::getActive(Bool* active)
-{
-    return daqTry([&]() {
+{   
+    try
+    {
         *active = this->template readValue<IBoolean>("Active");
-        return OPENDAQ_SUCCESS;
-    });
+    }
+    catch(...)
```

Comment:
In general, it was a request to not fail on connect, unless it's a critical error. Invalid nodes should simply be skipped and the error logged.

Reply:
Yes, but we need to be way more conservative about this. In this case we should catch only opcua communication / permission errors. 
- if node is not of correct type (as specified in TMS model), we should fail
- if a **mandatory** node is missing, we should fail
It is the only way of how to enforce correct device structure.

Reply:
I don't really agree.

Basically the request was to make the client-side of OPC UA more lenient towards smaller mistakes/errors on the serve-side model design. This to me means that on connect we should not fail when the structure of the device is slightly malformed - we should just ignore the malformed nodes.

As for function calls - those need to be evaluated on an individual basis. For me, `getName` for example should not fail, but just return the local ID, as that's good enough. Some other calls, on the other hand, should maybe throw an error.

The structure on our side will be enforced no matter what - if the server has a malformed part of the tree, that part will simply not appear on the client side.

I do agree though that we could go and review all the try catches here, and possibly sometimes log and rethrow.

Reply:
I do agree about the getName(). However we should still not be wrapping everything in try catch. 

In my opinion we should only skip objects at component level. Or maybe property property level at most. That means that if a component is missing a mandatory node, we should skip the whole component. Of course, we can try to assume the default value for that node if it makes sense, but in most cases, we should skip the whole component.

Reply:
I mean, I'd think we skip it at the bottom-most level. If a device is missing a component, we should just skip the component. If a property has missing fields, we can probably just revert to defaults - the server should handle things like "readOnly" anyhow. 

I think the intended philosophy was to be more forgiving on connect, to parse as much of the structure as possible, even if the server does not 100% adhere to all modelling rules (eg. some nodes are null). Function calls should probably still throw errors though. 

We decide to log + rethrow now, so that the rethrow can be removed at any point if deemed necessary.

Personally, I think this PR is an improvement over the previous state where throwing/logging was inconsistently handled, and should solve a potential issues on the HBK integration side.